Feat/dv 943 reown starterkit (#281)
rsksmart/devportal: docs/02-developers/04-quickstart/reown.md

> ---

Reviewer: Please add Reown as a tool under devtools --> index.md. Search Wagmi Starter Kit and use the same values `demos, src` for the Reown Starter Kit.

Also add the reown kit in index section of quick starts

Author: Noted, and added

sidebar_label: Reown
sidebar_position: 108
title: Rootstock Reown-Wagmi Starter Kit
description: 'The Rootstock Reown-Wagmi Starter Kit provides a solid foundation for developing decentralized applications (dApps) on the Rootstock blockchain. It streamlines development by leveraging React, Reown, Wagmi, and Shadcn libraries.'
tags: [rsk, rootstock, developers, reown, wagmi, quickstart, dApps, Smart Contracts]
---

The Rootstock Reown-Wagmi starter kit provides a foundation for building decentralized applications (dApps) on the Rootstock blockchain. 
It leverages the security of Bitcoin and the flexibility of Ethereum. 

> The kit uses [Reown](https://reown.com/) (previously WalletConnect) to handle wallet management, [Wagmi](https://wagmi.sh/), a React Hooks library, to simplify smart contracts and blockchain network interactions, and [Shadcn libraries](https://ui.shadcn.com/). 

Reviewer 2: There is no context about shadcn lib.   I will add at the end:

and Shadcn libraries, a set of customizable and accessible UI components for React, designed to streamline frontend development

This is kind of the same as you did with wagmi ;)

      ```
    - **npm:** npm comes bundled with the Node.js installation. To verify your npm installation, run:
      ```bash
      npm -v
      ```
      If you need to update npm to the latest version, you can run:
      ```bash
      npm install -g npm@latest
      ```
- **Basic Knowledge:** 
    - [React](https://react.dev/) (a JavaScript library for building user interfaces)
    - [Solidity](https://soliditylang.org/) (a programming language for Ethereum smart contracts).

:::tip[Rootstock Blockchain Developer Course]

Learn how to write, test, secure, deploy, and verify smart contracts on the Rootstock blockchain network. Enroll in the [Rootstock Blockchain Developer Course](/resources/courses/).
:::

## Setup

### 1. Clone the Repository
First, you’ll need to clone the Rootstock Reown Starter Kit repository. Open your terminal and run the following commands:
```bash
git clone https://github.com/rsksmart/rsk-reown-starter-kit
cd rsk-reown-starter-kit
```

### 2. Get Project ID
Every dApp that relies on Reown (previously WalletConnect) now needs to obtain a project ID from [Reown Cloud](https://cloud.reown.com). This is free and only takes a few minutes.

To get the key:

> 1. Go to [Reown](https://cloud.reown.com/sign-in) and sign up.

Reviewer: Please mention reown in the prerequisites

Reviewer 2: No need to, in next steps when the user runs yarn after cloning the project, it will be installed as one of the dependencies

Reviewer 2: take a look to line 72

2. Create a new project by clicking on **Create Project**.

> 3. Add a name and link to your project, select a product (AppKit or WalletKit), and select **WalletKit**.

Reviewer 2: you mentioned the selection twice.
It should be:
3. Add a name and link to your project,  on the product selection page between AppKit and WalletKit, select **WalletKit**

> 4. Now you will see the project ID. Copy it.

Reviewer 2: it sounds better with:
4. Your project ID is shown in the left menu under your project name. Click to copy it

> 3. Add a name and link to your project, on the product selection page between AppKit and WalletKit, select **WalletKit**.

Reviewer: ```suggestion
3. Add a name and link to your project, on the product selection screen, select **WalletKit** and continue.
```

### 3. Environment Setup
To set up your environment, follow these steps:

>   - Create a `.env` file and add environment variables.

Reviewer: Update the commented out text to reown project id in the starter kit

Reviewer 2: done
https://github.com/rsksmart/rsk-reown-starter-kit/pull/15

    ```text
    VITE_WC_PROJECT_ID=Your project ID from Reown Cloud
    VITE_BUNDLER_API_KEY='etherspot_public_key'
    VITE_CUSTOM_BUNDLER_URL=https://rootstocktestnet-bundler.etherspot.io/
    ```

>   - Enter your project ID. For testnet purposes, you can keep the Etherspot Bundler API key and Bundler URL. If you need a production environment, please go to [Etherspot](https://etherspot.io/), create an account, and obtain your API key.

Reviewer: Was Etherspot used? If not we can remove the env variables for Etherspot or show how to get it if needed.

Reviewer 2: yes, it is.   We used the abstract account feature

### 4. Install Dependencies
Before running the project, make sure to have the necessary dependencies installed. You can use NPM or Yarn. Run the following command to install dependencies:
```bash
yarn
```

### 5. Run the Project
Now that you’ve cloned the repository and installed dependencies, it’s time to run the project. Execute the following command:
```bash
yarn dev
```

> This will start the Rootstock Reown Starter DApp locally, allowing you to develop and test your smart contracts. You can access the Vite server at [http://localhost:5173](http://localhost:5173).

Reviewer 2: ```suggestion
This will start the Rootstock Reown Starter dApp locally, allowing you to develop and test your smart contracts. You can access the Vite server at [http://localhost:5173](http://localhost:5173).
```

## Result

<img src="/img/developers/quickstart/reown-starter-kit/1.reown.png"  width="800" height="600"/>

:::info[Info]

After successfully running your project using the command above, do the following: 

- Click the “Connect” button to log in. Once connected, you can:
- **Switch Networks:** Easily switch between Mainnet and Testnet.
- **View and Copy Your Address:** Access your wallet address.
- **Check Your tRBTC Balance:** See your tRBTC balance.
- **Disconnect:** Log out from the project.

:::

## Test Project
To test the project, follow these simple steps:
1. **Connect Your Wallet:** Click the “Connect” button.
2. **Navigate to the Reown-Wagmi Section:** Scroll down and find the card labeled “Contract Interaction with Reown Starter Kit.” Click it.
3. **Explore the Tabs:** In the Wagmi section, you’ll see three tabs: ERC-20, ERC-721, and ERC-1155. Click on any of these tabs to explore further.

<img src="/img/developers/quickstart/reown-starter-kit/2.reown.png"  width="800" height="600"/>

## Understanding the Codebase

### Folder Structure

```

> Public

Reviewer: ```suggestion
public
src
```
